engine/annotations/openpose.py

```diff
@@ -0,0 +1,255 @@
+        name="Bone Side",
+        items=((str(s.value), s.name.title(), '') for s in Side)
+    )
+
+ArmatureOpenPoseData = type('ArmatureOpenPoseData', (bpy.types.PropertyGroup,), {
+    "bl_label": "OpenPose",
+    "bl_idname": "dream_textures.ArmatureOpenPoseData",
+    "__annotations__": { b.name: bpy.props.BoolProperty(name=b.name.title(), default=True) for b in Bone },
+})
+
+def render_openpose_map(context, collection=None):
+    e = threading.Event()
+    result = None
+    def _execute():
+        nonlocal result
+        width, height = context.scene.render.resolution_x, context.scene.render.resolution_y
+        offscreen = gpu.types.GPUOffScreen(width, height)
+
+        with offscreen.bind():
+            fb = gpu.state.active_framebuffer_get()
+            fb.clear(color=(0.0, 0.0, 0.0, 0.0))
+            gpu.state.depth_test_set('LESS_EQUAL')
+            gpu.state.depth_mask_set(True)
+
+            lines = {
+                (Bone.NOSE, Bone.CHEST): (0, 0, 255),
+                (Bone.CHEST, Bone.SHOULDER_L): (255, 85, 0),
+                (Bone.CHEST, Bone.SHOULDER_R): (255, 0, 0),
+                (Bone.SHOULDER_L, Bone.ELBOW_L): (170, 255, 0),
+                (Bone.SHOULDER_R, Bone.ELBOW_R): (255, 170, 0),
+                (Bone.ELBOW_L, Bone.HAND_L): (85, 255, 0),
+                (Bone.ELBOW_R, Bone.HAND_R): (255, 255, 0),
+                (Bone.CHEST, Bone.HIP_L): (0, 255, 255),
+                (Bone.CHEST, Bone.HIP_R): (0, 255, 0),
+                (Bone.HIP_L, Bone.KNEE_L): (0, 170, 255),
+                (Bone.HIP_R, Bone.KNEE_R): (0, 255, 85),
+                (Bone.KNEE_L, Bone.FOOT_L): (0, 85, 255),
+                (Bone.KNEE_R, Bone.FOOT_R): (0, 255, 170),
+                (Bone.NOSE, Bone.EYE_L): (255, 0, 255),
+                (Bone.NOSE, Bone.EYE_R): (85, 0, 255),
+                (Bone.EYE_L, Bone.EAR_L): (255, 0, 170),
+                (Bone.EYE_R, Bone.EAR_R): (170, 0, 255),
+            }
+                                
+            with gpu.matrix.push_pop():
+                ratio = width / height
+                projection_matrix = mathutils.Matrix((
+                    (1 / ratio, 0, 0, 0),
+                    (0, 1, 0, 0),
+                    (0, 0, -1, 0),
+                    (0, 0, 0, 1)
+                ))
+                gpu.matrix.load_matrix(mathutils.Matrix.Identity(4))
+                gpu.matrix.load_projection_matrix(projection_matrix)
+                gpu.state.blend_set('ALPHA')
+
+                def transform(x, y):
+                    return (
+                        (x - 0.5) * 2 * ratio,
+                        (y - 0.5) * 2
+                    )
+
+                shader = gpu.shader.from_builtin('3D_UNIFORM_COLOR')
+                batch = batch_for_shader(shader, 'TRI_STRIP', {"pos": [(-ratio, -1, 0), (-ratio, 1, 0), (ratio, -1, 0), (ratio, 1, 0)]})
+                shader.uniform_float("color", (0, 0, 0, 1))
+                batch.draw(shader)
```

Comment:
IDK why but the first batch/draw is always coloring incorrectly and very transparent.

Before:
![openpose](https://user-images.githubusercontent.com/47096043/229312657-d4215112-b13b-4f0a-be40-488e3c8aeaaf.png)

After:
![openpose color fix](https://user-images.githubusercontent.com/47096043/229312677-28ffbd7e-e36c-4067-8666-3a62816a2ca5.png)

```suggestion
                shader = gpu.shader.from_builtin('3D_UNIFORM_COLOR')
                batch = batch_for_shader(shader, 'TRI_STRIP', {"pos": [(-ratio, -1, 0), (-ratio, 1, 0), (ratio, -1, 0), (ratio, 1, 0)]})
                shader.uniform_float("color", (0, 0, 0, 1))
                batch.draw(shader)

                batch = batch_for_shader(shader, 'TRI_STRIP', {"pos": [(-ratio, -1, 0), (-ratio, 1, 0), (ratio, -1, 0), (ratio, 1, 0)]})
                shader.uniform_float("color", (0, 0, 0, 1))
                batch.draw(shader)
```